content/utmt/rooms/creating new rooms/design/_index.md

```diff
@@ -0,0 +1,102 @@
+{{<hint type="warning">}}
+Never delete objects, or basically anything in UTMT. Everything in UTMT is organized in a large list, with the newest addition always located in the bottom. Deleting something that is not the last thing in its list can have unforseen consequences and will almost always break things. If you mistakenly put the wrong object in your room, replace it with another object that you actually need (by dragging the new object into the definition box of your misplaced object).
+{{</hint>}}
+
+{{< img name="initialCollision" size=origin >}}
+
+For puzzles and other uses, you can also search for `missile`, `bomb`, `smissile` and other destroyable blocks. Keep in mind that these blocks, as well as solid blocks, are invisible and do not appear without putting tiles over them or giving them creation code. 
+For this, there are two solutions:
+1. Add creation code for each of them and use `tile_link(tileset,x,y)` as its content. It will link the block to a tileset and automatically use a 16 x 16 size. The x and y coordinates correspond with the top-left corner of the breakable block.
+2. Place a tile in the position where the breakable block will be, and set the depth of that tile to either `-111` or `-112`.
+
+{{<hint type="tip">}}
+Here are all the breakable Blocks, along with what they are commonly refered to:
+- `oBlockShoot`; Beam blocks
+- `oBlockStep`; Crumble blocks
+- `oBlockBomb`; Bomb blocks
+- `oBlockMissile`; Missile blocks
+- `oBlockSMissile`; Super Missile Blocks
+- `oBlockSpeed`; Speedboost Blocks
+- `oBlockPBomb`; Power bomb blocks
+- `oBlockScrew`; Screw attack blocks
+Some breakables (namely Shoot, Bomb and PBomb) also have Chain Variants.
+{{</hint>}}
+
+{{< img name="breakablesSetup" size=origin >}}
+
+For backgrounds, you should place an `oBackground` object somewhere in the room. Usually they are placed in the top-left of the room. The `oBackground` needs `creation code` to function. Without it, the game will crash when trying to load the room. Right click on `Code` in the viewer on the left side and then click add. This will open a blank text window. As with rooms, object creation code follows a naming convention, starting with `gml_RoomCC_`, followed by the name of the room the object is located in (in our case it would be `rm_a1a13`), which is then followed by the code entry's ID (the number that automatically generates when you first add your code element, in our case `9808`) and finally `_Create`.
```

Comment:
```suggestion
For backgrounds, you should place an `oBackground` object somewhere in the room. Usually they are placed in the top-left of the room. The `oBackground` needs creation code to function. Without it, the game will crash when trying to load the room. Right click on `Code` in the viewer on the left side and then click `add`. This will open a blank text window. As with rooms, object creation code follows a naming convention, starting with `gml_RoomCC_`, followed by the name of the room the object is located in (in our case it would be `rm_a1a13`), which is then followed by the code entry's ID (the number that automatically generates when you first add your code element, in our case `9808`) and finally `_Create`.
```